.github/workflows/pr.yaml

```diff
@@ -10,14 +10,29 @@ on:
   pull_request:
 
 jobs:
-  pre-integration-checks:
+  pre-integration-checks-juju2:
     secrets: inherit
     uses: ./.github/workflows/pre_integration_checks.yaml
+    with:
+      libjuju-version-specifier: ==2.9.42.4
 
-  integration:
+  integration-juju2:
     needs:
-      - pre-integration-checks
+      - pre-integration-checks-juju2
     uses: ./.github/workflows/integration.yaml
     with:
       juju-snap-channel: 2.9/stable
       libjuju-version-specifier: ==2.9.42.4
+
+  pre-integration-checks-juju3:
+    secrets: inherit
+    uses: ./.github/workflows/pre_integration_checks.yaml
```

Comment:
To limit tests in a matrix you probably can use something like https://github.com/canonical/postgresql-operator/blob/a222afe209cc1fc129ff6edd31b51917fe9b9d86/.github/workflows/ci.yaml#L92, which (in the example) makes one test run only on Juju 2.

Reply:
I see your point and indeed `include`/`exclude` could be a very elegant solution here.

The problem is that it can only use matrix variables -- not `input`. I mean we'd need to do a major restructuring.
Which of course is possible - the question whether it's worth it.

I'm wondering a bit if we could turn the whole `tox-environment` list into an `input` list... Which technically seems possible, not very elegant tho: https://stackoverflow.com/questions/69781005/combine-dynamic-github-workflow-matrix-with-input-values-and-predefined-values

And in both cases there's the "outlier" `h-scaling-integration`....

I'd like to make this prettier, but I'm really not sure how :-(